From f41190552f44963ea586470280053792a169a694 Mon Sep 17 00:00:00 2001
From: Jeremy List <jeremylist@thunix.net>
Date: Thu, 1 Aug 2024 14:29:03 +1200
Subject: [PATCH] package: make aws-sdk-cpp build input optional

I have added an option to turn off this build input because I'm much
more comfortable when I don't have that type of thing on my computer.
Its default value is true in order to avoid impacting anyone who depends
on AWS features.

Change-Id: Ic57f3c9b9468f422e9fbdcf3ba0fe96177631067
---
 package.nix | 18 +++++++++++-------
 1 file changed, 11 insertions(+), 7 deletions(-)

diff --git a/package.nix b/package.nix
index 61015bac9..06f1107e0 100644
--- a/package.nix
+++ b/package.nix
@@ -80,13 +80,17 @@ let
   versionJson = builtins.fromJSON (builtins.readFile ./version.json);
   version = versionJson.version + versionSuffix;
 
-  aws-sdk-cpp-nix = aws-sdk-cpp.override {
-    apis = [
-      "s3"
-      "transfer"
-    ];
-    customMemoryManagement = false;
-  };
+  aws-sdk-cpp-nix =
+    if aws-sdk-cpp == null then
+      null
+    else
+      aws-sdk-cpp.override {
+        apis = [
+          "s3"
+          "transfer"
+        ];
+        customMemoryManagement = false;
+      };
 
   # Reimplementation of Nixpkgs' Meson cross file, with some additions to make
   # it actually work.
-- 
2.45.2

